### ISSUE TITLE
Rook ceph mon:  unable to stat setuser_match_path /var/lib/ceph/mon/ceph-a/store.db: (2) No such file or directory

### ISSUE THREAD

**Issue by ader1990** [rook/rook#14251]
There seems to be a concurrency issue in the pod/rook-ceph-mon-a: `mon` container and the `init-mon-fs` container do not behave in a way that is concurrency proofed. And the `mon` container cannot find the required rocksdb file the  `init-mon-fs` should have had created.

Rook version tested: 1.13 and 1,14 (multiple minor versions)
Ceph version tested: 18.2.0

```
kubectl logs -n rook-ceph pod/rook-ceph-mon-a-77974bc5f8-5xh2l
unable to stat setuser_match_path /var/lib/ceph/mon/ceph-a/store.db: (2) No such file or directory
```

If I delete the pod rook-ceph-mon-a-77974bc5f8-5xh2l, it gets recreated and everything works as expected, leading to my conclusion that this is a concurrency problem.

are there any rules in place so that `mon` container waits correctly for the `init-mon-fs` container to finish?

**Comment by ader1990** [comment]
I verified the rook-ceph-mon-a-77974bc5f8-5xh2l pod description and looks like the init containers `chown-container-data-dir` and `init-mon-fs` have been run before the `mon` container correctly. Still investigating what happens. What is certain is that if I remove the mon-a pod, another one gets created and the newly created pod's `mon` container runs succesfully.

**Comment by ader1990** [comment]
@travisn thank you for the confirmation, as I also stated in the previous comment, it does not seem like a concurrency issue anymore, as the other two containers already finished their initialisation. Maybe there s another issue with mount points from host to container, will have take a better look (trying now to properly replicate this issue in multiple envs).
Updated the issue name to properly reflect the current state of understanding.
